### DISCUSSION TITLE
D1 Mini Pro Issues [SOLVED]

### DISCUSSION THREAD

**Discussion by gatoruss** [forkineye/ESPixelStick#733] [General]
I am trying to load the espixelstick software to a D1 Mini Pro 16M.  I am having issues connecting to the network (and the device is not recognizing my micro SD module).

I have 2 builds of V4 -- one I downloaded in November 2023 and one I downloaded today (January 18, 2024).

Using the most recent build, and selecting D1 Mini Pro 16M as profile, the software appears to upload to the D1 Mini Pro, but the device cannot connect to my home network.  In fact, although I entered my SSID and password into the boxes in the installer, the device tries to connect to "MaRtInG" (which I assume to be Martin's home network).  After that fails several times, the device tries to connect as a SOFT AP -- unsuccessfully.  If I do the same thing with the build from November, I get a similar result, but instead of trying to connect to "MaRtInG," the device unsuccessfully tries to connect to "DEFAULT_SSID_NOT_SET."

If I use the installer to upload the software to the D1 Mini Pro using "Espixelstick" as the profile, the software loads and connects to my home network. 

I all 3 cases, however, the logs say that "No SD Card Installed."  However, I do have a micro SD module connected.  I have read some threads suggesting that this could be remedied by configuring the gpio pins on the Device's Set Up tab on the device's web browser; however, I accessed the browser Admin tab (when the device was programed using the  "Espixelstick" profile) and checked "Advanced," but I do not see any way to configure the GPOIs other than the ability to make some "active HIGH" of "active LOW."  See screen shot of the Device's Set Up tab.  

Any ideas?

[d1_mini_pro_16M.log](https://github.com/forkineye/ESPixelStick/files/13980471/d1_mini_pro_16M.log)
[d1_mini_pro_16M_(prior_distribution).log](https://github.com/forkineye/ESPixelStick/files/13980472/d1_mini_pro_16M_.prior_distribution.log)
[Espixelstick2.log](https://github.com/forkineye/ESPixelStick/files/13980474/Espixelstick2.log)
![Device_Set_Up_Screenshot 2024-01-18 124707](https://github.com/forkineye/ESPixelStick/assets/27308192/70899328-a68a-41b2-99e0-f0035473e51a)

**Comment by MartinMueller2003** [comment]
The active high/low is for the Button input pin. The SD Settings are above marked with the word SD. However, you seem to be missing three of the pins.

Please install the 16M version of the pro device using the latest dist.zip file. then go to the admin page and do a reset to factory defaults.
When the device finally enters the AP mode, open your tablet and set your wifi to the esp. 
Then open a browser on your tablet and go to the network setting page. 
enter your correct network settings and save the config.
reboot.
 You should connect to the proper network.
Send me a screenshot of the admin page and the device config page.

**Comment by MartinMueller2003** [comment]
The 16M log shows that the image did not upload properly. Also, My pro version requires an external antenna. Do you have your antenna connected?

**Comment by gatoruss** [comment]
Thank you for your quick reply, Martin.

> The 16M log shows that the image did not upload properly. 

My bad -- I "_**thought**_" that the log indicated that the image had in fact upload property?  I assumed that when it said "Done" after each step, that was indicative of success.  How do I remedy this?

>Also, My pro version requires an external antenna. Do you have your antenna connected?

I do not have an external antenna connected, but  I do not believe that I need to have an external antenna.  My version has a connection for an external antenna, but is also has a ceramic onboard antenna.  By default, the onboard antenna is enabled -- to use the external antenna it is necessary to re-position a 0 ohm resistor (i.e., disconnect the resistor from the onboard antenna and connect it to the external antenna).  I have not moved the resistor, and I have previously connected the very device to my network (just not when running expixelstick software using the D1 Mini Pro 16M profile).

**Comment by MartinMueller2003** [comment]
The log showed that some of the upload steps were out of order. That indicates a problem.

I only asked about the antenna because running a transmitter without an antenna can be bad for the transmitter.

**Reply by gatoruss** [reply]
Thank you for your help, Martin.  I have run into some unrelated technical issues -- I accidentally damaged the device in another experiment and need to acquire another d1 mini pro.  Once I have it I will pick-up where this thread left off.

I appreciate your assistance.

**Comment by gatoruss** [comment]
Okay.  I have another d1 mini pro that has not been used before.  I loaded the d1 Mini pro 16M firmware using the most recent distribution.  Initially, the installer said that it could not erase flash and to make sure the device was in programming mode.

Here is what I did:

I used the arduino ide to erase the flash uploaded blank "bare minimum" sketch.  

I was then able to upload the espixelstick program using the most recent installer.  However, the device would not connect to my network (as was the case with the prior device -- see original post).   It then tried unsuccessfully to connect to SOFT AP, and continued to re-boot numerous times.  The attached log file entitled "d1_mini_pro_16M_(19JAN2024)__01" shows what I am describing.  

Then I searched the networks that were available to my laptop and I saw an AP for the device and I connected to it.  The serial monitor than reported that the device was connected as an AP with an IP of 192.168.4.1.  But when I tried to access that address, I was told that "Page not found" (see attached screen shot).  The attached log file entitled "d1_mini_pro_16M_(19JAN2024)__02" shows what I am describing.

Note, I was subsequently able to successfully load WLED on this device.

[d1_mini_pro_16M_(19JAN2024)__01.log](https://github.com/forkineye/ESPixelStick/files/13990691/d1_mini_pro_16M_.19JAN2024.__01.log)
[d1_mini_pro_16M_(19JAN2024)__02.log](https://github.com/forkineye/ESPixelStick/files/13990692/d1_mini_pro_16M_.19JAN2024.__02.log)
![Screenshot 2024-01-19 093605](https://github.com/forkineye/ESPixelStick/assets/27308192/f65fa567-c234-4816-909b-3b572a27dc21)

**Comment by MartinMueller2003** [comment]
The page not found indicates that the file system did not get initialized. This in the log confirms that the FS did not get set up:
09:28:37: [FileMgr] Flash file system initialized.
09:28:37: [FileMgr] Listing directory: /
09:28:37: [FileMgr] No SD card installed

I will need to verify that the addresses are correct. The flash log shows that an FS was uploaded but it looks like the application cannot find it.

**Comment by MartinMueller2003** [comment]
It looks like the mini pro 16 configuration is trash. However, the d1_mini_pro config looks like it is working. I am removing the 16M config since it is not needed.

**Comment by MartinMueller2003** [comment]
Dist.zip files have been updated.

**Comment by gatoruss** [comment]
I downloaded new dist.zip and programmed device with installer (selecting D1 Mini Pro as profile).

The installer uploaded the firmware and the device connected to my network.

I  accessed the device and checked advance on the admin tab.  However, when I go to the device Setting tab and select  FPP Remote for secondary input, the "FSEQ File To Play" drop down has no selections and the GPOI pins can only be set to active hi or active lo,  No other pins can be set.  Screen shot below.

![Screenshot Device Setup](https://github.com/forkineye/ESPixelStick/assets/27308192/2486da88-a282-4a0d-b92e-891018cec437)

**Comment by MartinMueller2003** [comment]
You wont get any file choices until an SD card is detected. You wont get an SD card present until you set the GPIO values for the SD card. Three out of the four settings are missing on the UI. I found the issue and have uploaded a new set of dist.zip files.

**Reply by gatoruss** [reply]
Follow-up question.  I have the sd card module connected to D1 mini pro using the pins indicated on the device setup page (the default pins on the device set up page appear correct based on the pinout for the device.

However, when boot up the device and select FPP Remote as "Secondary Input," the FSEQ File to Play drop down menu is empty.  I assume that the device is not recognizing the SD Card.

Any suggestions?

**Comment by gatoruss** [comment]
I downloaded the new dist.zip. 
Installed firmware on device.
Device accessed network quickly.
I accessed device thru browser, and checked box on admin tab (screen shot below).  But device page still shows only "SD Chip Select Pin," still just 3/4 pins (see screen shot).
Installation log attached.
![Screenshot admin Tab](https://github.com/forkineye/ESPixelStick/assets/27308192/2c0e2b05-f6a6-44d4-8a06-270c6dc0fa0d)
![Screenshot Device Setup 02](https://github.com/forkineye/ESPixelStick/assets/27308192/c4fad9d4-c5c5-42e0-9298-0dd9c793e160)
[log_espixell_install_20Jan2024.log](https://github.com/forkineye/ESPixelStick/files/13998751/log_espixell_install_20Jan2024.log)

**Comment by MartinMueller2003** [comment]
Not sure where you got that dist file but it was not from one of my two locations. That is the beta 5 build from last November.

**Comment by gatoruss** [comment]
I apologize.  
I downloaded your most recent dist.zip folders, but must have gotten it confused with a prior download (from last November) when extracting the zip.
I installed the firmware with your most recent dist.zip, and it appears to work like a charm.  Screen shot below.
in case you needed it for any reason, the  install log is attached.
Thank you very much for working thru this with me.  I appreciate your time.
[log_espixell_install_20Jan2024_02.log](https://github.com/forkineye/ESPixelStick/files/13998823/log_espixell_install_20Jan2024_02.log)
![Screenshot Device Setup 03](https://github.com/forkineye/ESPixelStick/assets/27308192/8db64ca5-9590-452d-9d57-fb7793abd1e0)

**Comment by MartinMueller2003** [comment]
No problem. Glad you are up and running. Have fun.

**Comment by MartinMueller2003** [comment]
When you look at the menu tabs do you see the File Administration tab? If not, the SD card is not being detected and that list will be empty.
Leave the flashtool connected and log a boot. Post the log here.
Martin
    On Monday, January 22, 2024 at 02:58:06 PM EST, gatoruss ***@***.***> wrote:  
 
 


Follow-up question. I have the sd card module connected to D1 mini pro using the pins indicated on the device setup page (the default pins on the device set up page appear correct based on the pinout for the device.

However, when boot up the device and select FPP Remote as "Secondary Input," the FSEQ File to Play drop down menu is empty. I assume that the device is not recognizing the SD Card.

Any suggestions?

**Comment by gatoruss** [comment]
No, there is no File Administration tab; just an Admin tab.  See screen shot.

Log attached.

Thank you.

![Screenshot D1MiniPro espixel 01](https://github.com/forkineye/ESPixelStick/assets/27308192/1361699d-423f-44e9-9711-8095852e8a79)

[log_espixell_boot_23Jan2024_01.log](https://github.com/forkineye/ESPixelStick/files/14015143/log_espixell_boot_23Jan2024_01.log)

**Comment by MartinMueller2003** [comment]
This line in the log says it all. It means that the SD Card has not been connected properly.

15:25:08: [FileMgr] No SD card installed

What type of SD interface are you using? How is it connected to the ESP?
    On Monday, January 22, 2024 at 03:30:50 PM EST, gatoruss ***@***.***> wrote:  
 
 


No, there is no File Administration tab; just an Admin tab. See screen shot.

Log attached.

Thank you.

Screenshot.D1MiniPro.espixel.01.png (view on web)

log_espixell_boot_23Jan2024_01.log

**Comment by gatoruss** [comment]
>What type of SD interface are you using? 

I am using a HiLetgo micro sd module --> [SD Module](https://www.amazon.com/gp/product/B07BJ2P6X6/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1)

This the SD card I am using --> [SD card](https://www.amazon.com/gp/product/B07BJ2P6X6/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1)

>How is it connected to the ESP?

CS of module <--> GPIO 15 of device (d1 mini pro pin D8)
SCL of module <--> GPOI 14 of device (d1 mini pro pin D5)
MISO of module <--> GPOI 12 of device (d1 mini pro pin D6)
MOSI of module <--> GPIO 13 of device (d1 mini pro pin D7)

**Comment by gatoruss** [comment]
I am wondering if there is an issue with voltage.  

First, the SD module has a 3.3 voltage regulator onboard, converting Vcc from 5v to 3.3v.  I had assumed that this issue was addressed by powering the SD module with 5v from breadboard.

Second, the SD module has an onboard level converter, which is taking what it expects to be 5v signals from an Arduino and converts the signal to 3.3v; however, the signals from the ESP8266 are already at 3.3v.  if this is the issue, I may need a different SD module.

**Comment by gatoruss** [comment]
UPDATE:  I am updating this thread in case someone else has a similar issue.  

Seem like the voltage was the issue.  I purchased this micro SD module --> [https://www.amazon.com/dp/B0989SM146?psc=1&ref=ppx_yo2ov_dt_b_product_details](url).  Note, although the information at the link states the module works at 3.3 or 5v, that is in correct as it is 3.3 only -- but works with esp8266. 

When connected to module, the d1 mini pro recognizes the module, the File Administration tab is available (see screen shot below) and on the Device Setup tab, with Remote as "Secondary Input" selected,  the FSEQ File to Play drop down menu shows expected files (see screen shot below).
![D1_mini_pro_device setup tab](https://github.com/forkineye/ESPixelStick/assets/27308192/33ad9bd0-d163-46dc-9a69-19f9b212721d)
![D1_mini_pro_file_management_tab](https://github.com/forkineye/ESPixelStick/assets/27308192/18398839-3fd4-4bb2-8157-fc7c4bce9958)
